api/src/migrations/geoprocessing/1611221157285-initialGeoDBSetup.ts

```diff
@@ -0,0 +1,185 @@
+        "modify_at" timestamp NOT NULL default now()
+      );
+
+      -- Features data table (will hold species and bioma data).
+      CREATE TABLE "features_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "the_geom" geometry CONSTRAINT features_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "properties" jsonb,
+        "source" source_type
+      );
+
+      -- Usage of features data within scenarios table (will hold species and bioma data).
+      CREATE TABLE "scenario_features_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "feature_class_id" uuid,
+        "scenario_id" uuid,
+        "total_area" varchar,
+        "admin_area" timestamp,
+        "current_pa" source_type,
+        "spf" float8,
+        "target" float8,
+        "target2" float8,
+        "targetocc" float8,
+        "sepnum" float8,
+        "created_at" timestamp NOT NULL default now(),
+        "created_by" uuid NOT NULL,
+        "modify_at" timestamp NOT NULL default now()
+      );
+
+      -- Planning units geometry data table.
+      CREATE TABLE "planing_units_geom" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "the_geom" geometry UNIQUE NOT NULL CONSTRAINT features_geometry_valid_check CHECK (ST_IsValid(the_geom)),
+        "type" shape_type,
+        "size" int
+      ) PARTITION BY LIST (type);
+
+      CREATE TABLE planing_units_geom_square PARTITION OF planing_units_geom FOR VALUES IN ('square');
+      CREATE TABLE planing_units_geom_hexagon PARTITION OF planing_units_geom FOR VALUES IN ('hexagon');
+      CREATE TABLE planing_units_geom_irregular PARTITION OF planing_units_geom FOR VALUES IN ('irregular');
+
+      -- Planning units usage by scenario.
+      CREATE TABLE "scenarios_pu_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "pu_geomid" uuid NOT NULL REFERENCES "planing_units_geom" ("id"),
+        "scenario_id" uuid NOT NULL,
+        "puid" int NOT NULL,
+        "lockin_status" int,
+        "xloc" float8,
+        "yloc" float8,
+        "protected_area" float8
+      );
+
+      -- Cost data for planning units usage by scenario.
+      CREATE TABLE "scenarios_pu_cost_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "output_results_data_id" uuid NOT NULL REFERENCES "scenarios_pu_data" ("id"),
+        "cost" float8
+      );
+
+      CREATE TABLE "output_results_data" (
+        "id" uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+        "puid" int REFERENCES "scenarios_pu_data" ("puid"),
+        "scenario_id" uuid,
+        "run_id" uuid,
+        "value" float8,
+        "missing_values" jsonb
+      );
+
+
+      COMMENT ON TABLE "admin_regions" IS 'will cover adm0,adm1 and adm2';
+      COMMENT ON TABLE "features_data" IS 'For geometry storage';
+      COMMENT ON COLUMN "features_data"."properties" IS 'Either have the properties as a jsonb or as columns on this tables';
```

Comment:
do we need this comment in the db? it looks more like something that could belong to a design document - I like the idea of having properties as a JSONB to start with as in this migration, and we could revisit this (or split out of the JSONB properties that work best as columns) later on.